mlfinlab/feature_importance/importance.py

```diff
@@ -114,9 +125,10 @@ def mean_decrease_accuracy(model, X, y, cv_gen, sample_weight_train=None, sample
             fold_metrics_values.loc[i] = scoring(y.iloc[test], pred, sample_weight=sample_weight_score[test])
 
         # Get feature specific metric on out-of-sample fold
-        for j in X.columns:
+        for j in feature_sets:
             X1_ = X.iloc[test, :].copy(deep=True)
-            np.random.shuffle(X1_[j].values)  # Permutation of a single column
+            for j_i in j:
+                np.random.shuffle(X1_[j_i].values)  # Permutation of a single column for MDA or through the whole subset for CFI
```

Comment:
With random_state, we should create numpy Random State object and call rs_obj.shuffle

Reply:
Will be done.